### PR TITLE
Added authorization for VariableReference fields

### PR DESCRIPTION
**Pull request by bogdancice**
This is a fix for: https://github.com/graphql-dotnet/authorization/issues/27

### CODE REVIEW

**Comment by bogdancice** [comment]
@joemcbride can you please check this PR? we want to go in production in the next weeks and we can't go without this authorization

**Comment by mgkeen** [comment]
Any update on this? I'm running into the same issue.

File: src/GraphQL.Authorization.Tests/AuthorizationValidationRuleTests.cs [hunk 1]
```diff
@@ -178,6 +178,53 @@ public void Issue61()
         }
 
         [Fact]
+        public void passes_with_claim_on_variable_type()
+        {
+            Settings.AddPolicy("FieldPolicy", _ =>
+            {
+                _.RequireClaim("admin");
+            });
```
Comment (sungam3r) [inline src/GraphQL.Authorization.Tests/AuthorizationValidationRuleTests.cs:186]:
```suggestion
            Settings.AddPolicy("FieldPolicy", _ => _.RequireClaim("admin"));
```

File: src/GraphQL.Authorization.Tests/AuthorizationValidationRuleTests.cs [hunk 2]
```diff
@@ -178,6 +178,50 @@ public void Issue61()
...
+        }
+
+        [Fact]
+        public void fails_on_missing_claim_on_variable_type()
+        {
+            Settings.AddPolicy("FieldPolicy", _ =>
+            {
+                _.RequireClaim("admin");
+            });
```
Comment (sungam3r) [inline src/GraphQL.Authorization.Tests/AuthorizationValidationRuleTests.cs:209]:
```suggestion
            Settings.AddPolicy("FieldPolicy", _ => _.RequireClaim("admin"));
```

**Comment by sungam3r** [comment]
Thanks @bogdancice . Closed in favor of #179. I changed code a bit. Will be released in v4.1.

**Pull request closed without merging**